storage: TestConditionalPutUpdatesTSCacheOnError failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/5f1d58bc79224fa7b83645fc8822d137a83e6fd2

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestConditionalPutUpdatesTSCacheOnError PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1345149&tab=buildLog

```
=== RUN   TestConditionalPutUpdatesTSCacheOnError
I190618 06:12:31.241327 16741 util/stop/stopper.go:544  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190618 06:12:31.241351 16741 util/stop/stopper.go:544  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
--- FAIL: TestConditionalPutUpdatesTSCacheOnError (0.01s)
    replica_test.go:2826: expected write timestamp to upgrade to 2.000000000,1; got 2.000000000,2
    replica_test.go:2870: expected write timestamp to upgrade to 2.000000000,1; got 2.000000000,2

```

SHA: https://github.com/cockroachdb/cockroach/commits/d2818382eea5612d3d3e3263941456e0734776e3

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestConditionalPutUpdatesTSCacheOnError PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1348344&tab=buildLog

```
=== RUN   TestConditionalPutUpdatesTSCacheOnError
I190619 06:34:36.435891 16743 util/stop/stopper.go:544  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190619 06:34:36.435922 16743 util/stop/stopper.go:544  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
--- FAIL: TestConditionalPutUpdatesTSCacheOnError (0.01s)
    replica_test.go:2826: expected write timestamp to upgrade to 2.000000000,1; got 2.000000000,2
    replica_test.go:2870: expected write timestamp to upgrade to 2.000000000,1; got 2.000000000,2

```

--- a/pkg/storage/replica_test.go
+++ b/pkg/storage/replica_test.go
@@ -2807,7 +2807,7 @@ func TestConditionalPutUpdatesTSCacheOnError(t *testing.T) {
 	// CPut args which expect value "1" to write "0".
 	key := []byte("a")
 	cpArgs1 := cPutArgs(key, []byte("1"), []byte("0"))
-	_, pErr := tc.SendWrapped(&cpArgs1)
+	_, pErr := tc.SendWrappedWith(roachpb.Header{Timestamp: t2}, &cpArgs1)
 	if cfErr, ok := pErr.GetDetail().(*roachpb.ConditionFailedError); !ok {
 		t.Errorf("expected ConditionFailedError; got %v", pErr)
 	} else if cfErr.ActualValue != nil {
@@ -2895,7 +2895,7 @@ func TestInitPutUpdatesTSCacheOnError(t *testing.T) {
 
 	// InitPut args to write "1" to same key. Should fail.
 	ipArgs2 := iPutArgs(key, []byte("1"))
-	_, pErr = tc.SendWrapped(&ipArgs2)
+	_, pErr = tc.SendWrappedWith(roachpb.Header{Timestamp: t2}, &ipArgs2)
 	if cfErr, ok := pErr.GetDetail().(*roachpb.ConditionFailedError); !ok {
 		t.Errorf("expected ConditionFailedError; got %v", pErr)
 	} else if valueBytes, err := cfErr.ActualValue.GetBytes(); err != nil {

